app/src/play/java/de/danoeh/antennapod/dialog/CustomMRControllerDialog.java

```diff
@@ -124,38 +165,89 @@ public View onCreateMediaControlView(Bundle savedInstanceState) {
             }
         };
 
-        artView = new ImageView(getContext()) {
-            @Override
-            protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
-                int desiredHeight = heightMeasureSpec;
-                if (MeasureSpec.getMode(heightMeasureSpec) != MeasureSpec.EXACTLY) {
-                    Drawable drawable = getDrawable();
-                    if (drawable != null) {
-                        int originalWidth = MeasureSpec.getSize(widthMeasureSpec);
-                        int intrHeight = drawable.getIntrinsicHeight();
-                        int intrWidth = drawable.getIntrinsicWidth();
-                        float scale;
-                        if (intrHeight*16 > intrWidth*9) {
-                            // image is taller than 16:9
-                            scale = (float) originalWidth * 9 / 16 / intrHeight;
-                        } else {
-                            // image is more horizontal than 16:9
-                            scale = (float) originalWidth / intrWidth;
+        LinearLayout.LayoutParams artParams;
+        if (landscape) {
+            artView = new ImageView(getContext()) {
+                @Override
+                protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+                    int desiredWidth = widthMeasureSpec;
+                    int desiredMeasureMode = MeasureSpec.getMode(heightMeasureSpec) == MeasureSpec.EXACTLY ?
+                            MeasureSpec.EXACTLY : MeasureSpec.AT_MOST;
+                    if (MeasureSpec.getMode(widthMeasureSpec) != MeasureSpec.EXACTLY) {
+                        Drawable drawable = getDrawable();
+                        if (drawable != null) {
+                            int intrHeight = drawable.getIntrinsicHeight();
+                            int intrWidth = drawable.getIntrinsicWidth();
+                            int originalHeight = MeasureSpec.getSize(heightMeasureSpec);
+                            if (intrHeight < intrWidth) {
```

Comment:
I think this part needs more explanation (comments).
If this is something one can read up on, say, stack overflow, a link is sufficient.